app/src/main/java/com/example/playlistmaker/feature/search/ui/SearchActivity.kt

```diff
@@ -145,53 +120,55 @@ class SearchActivity : AppCompatActivity() {
         }
     }
 
-    private fun setupHistory() {
-        clearHistoryButton.setOnClickListener {
-            historyInteractor.clearSearchHistory()
-            showSearchHistory()
-        }
-    }
-
     private fun setupObservers() {
         viewModel.searchState.observe(this) { state ->
             when (state) {
-                is SearchStateUi.Loading -> showLoading()
-                is SearchStateUi.Success -> showResults(state.tracks)
-                is SearchStateUi.Error -> showError(state.message)
-                SearchStateUi.Empty -> showEmptyResults()
+                SearchUiState.Initial -> showInitialState()
+                SearchUiState.Loading -> showLoading()
+                is SearchUiState.Success -> showResults(state.tracks)
+                is SearchUiState.Error -> showError(state.message)
+                SearchUiState.Empty -> showEmptyResults()
+                is SearchUiState.History -> showSearchHistory(state.tracks)
             }
         }
     }
 
-    private fun onTrackClicked(track: Track) {
+    private fun onTrackClicked(track: com.example.playlistmaker.feature.search.domain.model.Track) {
         val currentTime = System.currentTimeMillis()
         if (currentTime - lastClickTime < CLICK_DEBOUNCE_DELAY) {
             return
         }
         lastClickTime = currentTime
 
-        historyInteractor.addTrackToHistory(track)
+        // Добавляем трек в историю через ViewModel
+        viewModel.addTrackToHistory(track)
 
         val intent = Intent(this, PlayerActivity::class.java)
         intent.putExtra("track", track)
         startActivity(intent)
     }
 
-    private fun showSearchHistory() {
-        val history = historyInteractor.getSearchHistory()
+    private fun showInitialState() {
+        historyLayout.visibility = View.GONE
```

Comment:
Рекомендую использовать экстеншн isVisible - он удобнее и читабельнее:

```suggestion
        historyLayout.isVisible = false
```